src/librustdoc/html/render/context.rs

```diff
@@ -183,8 +183,8 @@ impl<'tcx> Context<'tcx> {
         };
         title.push_str(" - Rust");
         let tyname = it.type_();
-        let desc = it.doc_value().as_ref().map(|doc| plain_text_summary(doc));
-        let desc = if let Some(desc) = desc {
+        let desc = plain_text_summary(&it.doc_value());
+        let desc = if !desc.is_empty() {
```

Comment:
This is a very slight behavior change, but I'm inclined to think of it as a probably-harmless bug fix.

Reply:
`plain_text_summary`'s special-casing is just for performance. It doesn't actually affect the result.

That said, I realize there's no actual change because `doc_value` doesn't have the empty docs edge case that `collapsed_doc_value` has.

Reply:
> plain_text_summary's special-casing is just for performance. It doesn't actually affect the result.

This is my point though: almost all the special casing of `doc_value()` is just for performance.

Reply:
I agree. I mistakenly thought this was one of the cases where removing the special-casing affected rustdoc's behavior.